### ISSUE TITLE
Selective CIDRAM

### ISSUE THREAD

**Issue by mikeruss1** [CIDRAM/CIDRAM#244]
Just thinking - I use php to serve my images as it gives me more control and hides the source folder.

However I get a lot of direct GETS unrelated to the original source page, I trap a lot of crap but some gets through. I dont put them through CIDRAM because of the increased load, maybe 10-15 images per page. 

But I could if you could vet the first one, then do exactly the same for subsequent GETS for the same source page. So if the referror is the same, the IP is the same, and its within 30 seconds (to allow for lazy loading), then do whatever you did for the first one.

My guess is the overhead would be very marginal.

This is a very specific mod which you might not be interested, quite understood.

What you think?

**Comment by 737simpilot** [comment]
Are the direct GETS from nefarious IPs? Are these IPs not blocked by CIDRAM? You said you 'trap a lot of crap', but I'm wondering if CIDRAM is doing that for the direct GET requests? Maybe users are hitting the pages via a search engine that's unrelated to the source page? CIDAM doesn't protect images, so from that perspective it's not going to work. But if the images are loaded via PHP CIDRAM will of course. Reason I say this is because the so-called increased load of images has no bearing on CIDRAM. Well, as far as I know. CIDRAM only protects PHP pages that I know of.

I wonder if the PHP that serves the images can be coded to allow only certain referrers based on the images and their path?

**Comment by mikeruss1** [comment]
Yes the <img src=> tag calls a PHP program for each image, so would be easy to call Cidram. I do some limited checks to ensure a session is enabled, and referrer is valid but cidram would do a better job if I could minimise load.

**Comment by Maikuolan** [comment]
The idea is good. I'll need to investigate the viability of it, though. Leave it with me for a bit and I'll reply back a bit later. :-)

**Comment by mikeruss1** [comment]
Logic seems similar to rate limiting so you may already have the basis?

**Comment by mikeruss1** [comment]
Another difficulty will be that it conflicts with rate limiting which would need to be turned off for images. Maybe getting over complex?

**Comment by mikeruss1** [comment]
maybe there is a much easier way to do things like this, and 237.
let the site decide when it wants to block something, then open an api for the site to tell CIDRAM to do something, like block future GETS for this ip address, start tracking, or whatever. Should be much easier for you and opens a standard interface between a site and CIDRAM.

**Comment by mikeruss1** [comment]
the API could be something like ...
IP 
UA
reason to show for block
action to take - maybe any or all of block/dont block, track/dont track, log/dont log, recaptcha on/off etc

much like auxiliary rules

**Comment by mikeruss1** [comment]
if you ever progress the API idea, could you also consider a sort of reverse API where CIDRAM notifies the app that the GET is from a known search engine? Maybe set some value in a global variable?

**Comment by Maikuolan** [comment]
> if you ever progress the API idea, could you also consider a sort of reverse API where CIDRAM notifies the app that the GET is from a known search engine? Maybe set some value in a global variable?

Internally, CIDRAM is already populating a `$CIDRAM['BlockInfo']['Verified']` variable, to be able to identify internally whether the request originates from a known search engine, so it would just be a matter of exposing that variable to any future API or related work. So yeah; Definitely should be doable. :-)
